src/pages/index.js

```diff
@@ -1,55 +1,160 @@
 function createCard(data) {
-  const card = new Card(data, ".card", () => {
-    imagePopup.open({ name: data.name, url: data.link });
+  const card = new Card(data, ".card", {
+    handleImageClick: () => {
+      imagePopup.open({ name: data.name, url: data.link });
+    },
+    deleteCallback: () => {
+      currentCard = card;
+      deleteCardPopup.getForm().querySelector(".form__submit").textContent =
+        "Delete";
+      deleteCardPopup.open();
+    },
+    likeCallback: () => {
+      return !card.isLiked
+        ? api.like({ cardId: card.getCard().id })
+        : api.unlike({ cardId: card.getCard().id });
+    },
   });
 
+  card.updateLike();
+
   return card.getCard().element;
 }
 
-const gallery = new Section(
-  {
-    items: cardList,
-    renderer: (data) => {
-      const cardElement = createCard(data);
-      gallery.addItem(cardElement);
-    },
-  },
-  ".gallery"
+// API
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  authToken: "7fef8830-65a5-4840-8327-1cd6adab3c4d",
+});
+
+// Render all initials information from API
+api.getUser().then((userData) => {
+  userInfo.setUserInfo({
+    name: userData.name,
+    job: userData.about,
+    id: userData._id,
+    pfpURL: userData.avatar,
+  });
+  userLoaded = true;
+});
+
+function renderCards() {
+  api.getCards().then((cards) => {
+    const gallery = new Section(
+      {
+        items: cards,
+        renderer: (item) => gallery.addItem(createCard(item), "append"),
+      },
+      ".gallery"
+    );
+
+    gallery.clear();
+    gallery.renderItems();
+  });
+}
+renderCards();
+
+// Change user info
+const editInfoPopupButton = document.querySelector(
+  ".profile__edit-name-button"
 );
 
+const userInfoPopup = new PopupWithForm(
+  { popupSelector: ".popup--edit-profile" },
+  (data) => {
+    userInfoPopup.getForm().querySelector(".form__submit").textContent =
+      "Saving...";
+    api.updateProfile({ name: data.name, about: data.job }).then((res) => {
+      userInfo.setUserInfo({ name: res.name, job: res.about });
+      userInfoPopup.close();
+    });
+  }
+);
+userInfoPopup.setEventListeners();
+editInfoPopupButton.addEventListener("click", () => {
+  userInfoPopup.setInputValues({
+    name: userInfo.getUserInfo().name,
+    job: userInfo.getUserInfo().job,
+  });
+  userInfoPopup.open();
+  userInfoPopup.getForm().querySelector(".form__submit").textContent = "Save";
+});
+
+// Add new card
+const newPlacePopupButton = document.querySelector(".profile__add-button");
 const newPlacePopup = new PopupWithForm(
   { popupSelector: ".popup--new-place" },
   (data) => {
-    gallery.addItem(
-      createCard({ name: data["place-name"], link: data["place-url"] })
-    );
+    newPlacePopup.getForm().querySelector(".form__submit").textContent =
+      "Creating...";
+    api
+      .addCard({
+        name: data["place-name"],
+        url: data["place-url"],
+      })
+      .then((res) => {
+        renderCards();
```

Comment:
`renderCards();` is not needed when you add a new card  because it's a waste of resources on fetching, waiting, making the server busy and so on



```js
        gallery.addItem(createCard(res))


```